server/static/js/uploadSticker.js

```diff
@@ -64,41 +64,46 @@ function previewImage(){
 }
 
 //Upload function
-function submit(){
+function submit() {
     latitudeInput.disabled = true;
     longitudeInput.disabled = true;
     submitButton.classList.add('addSubmitButtonPressed');
 
     //Create request
     var request = new XMLHttpRequest();
-    request.open('POST', 'upload', true);
-    //request.setRequestHeader('Content-type', 'application/x-www-form-urlencoded');
+    request.open('POST', '/upload', true);
+    request.withCredentials = true;
+    request.setRequestHeader('X-Requested-With', 'XMLHttpRequest');
+    request.onreadystatechange = function () {
+        if (this.readyState !== 4) return;
 
-    //Handle response
-    request.onreadystatechange = function() {
-        if(this.readyState == 4){
-            if(this.status == 200){
-                dropArea.classList.remove('invalid');
-                const response = this.responseText;
-                closeAddView();
-                resetView();
-                setTimeout(function(){
-                    openSuccessView();
-                    emailCode = JSON.parse(response)['emailCode'];
-                }, 500);
+        // detect redirect to login
+        if (this.status === 401 || (this.responseURL && this.responseURL.indexOf('/login') !== -1)) {
```

Comment:
I don't understand what `this` is. How is a code set here, what context happens before this that we can assume we might arrive here from a redirect?